src/system/SystemPacketBuffer.cpp

```diff
@@ -508,45 +508,63 @@ void PacketBuffer::AddRef()
 
 PacketBufferHandle PacketBufferHandle::New(size_t aAvailableSize, uint16_t aReservedSize)
 {
-    // Adding three 16-bit-int sized numbers together will never overflow
-    // assuming int is at least 32 bits.
+    // Ensure that SIZE_MAX is at least 32 bits
     static_assert(INT_MAX >= INT32_MAX, "int is not big enough");
+    static_assert(SIZE_MAX >= INT_MAX, "Our additions may not fit in size_t");
+
+    // Sanity check for kStructureSize and putting an upper bound.
     static_assert(PacketBuffer::kStructureSize == sizeof(PacketBuffer), "PacketBuffer size mismatch");
-    static_assert(PacketBuffer::kStructureSize < UINT16_MAX, "Check for overflow more carefully");
-    static_assert(SIZE_MAX >= INT_MAX, "Our additions might not fit in size_t");
-    static_assert(PacketBuffer::kMaxSizeWithoutReserve <= UINT32_MAX, "PacketBuffer may have size not fitting uint32_t");
+    static_assert(PacketBuffer::kStructureSize < UINT16_MAX, "kStructureSize should not exceed UINT16_MAX");
+    static_assert(PacketBuffer::kMaxSizeWithoutReserve <= UINT16_MAX, "PacketBuffer may have size not fitting uint16_t");
```

Comment:
I don't understand the message on this static assert....

Reply:
It may help to say __why__ this enforcement is done.  Right now it reads like `assert foo < 1234, "Foo should not exceed 1234"` which does not help users understand the sideffects of why this is being enforced. What happesn if I change the `UINT16_MAX` here with `UINT32_MAX` like it was before? why is uint16 max correct but uint32 max was wrong before?

Reply:
This is a relatively tight upper bound on the max buffer allocation for regular messages(i.e., not large) to ensure a corresponding upper bound on the sum, so that it does not overflow. UINT32_MAX is a much more loose upper bound for kMaxSizeWithoutReserve( which is equal to CHIP_SYSTEM_CONFIG_PACKETBUFFER_CAPACITY_MAX ).
Have added some text in the comment to qualify.